### DISCUSSION TITLE
With dag_discovery_safe_mode enabled, files with `Airflow` keyword are not scanned

### DISCUSSION THREAD

**Discussion by andrius-paurys** [apache/airflow#30206] [General]
### Apache Airflow version

2.5.2

### What happened

This files does not get recognized as a DAG file:
```
import airflow
import company.airflow.etlengine2 as engine

cli = engine.bootstrap(globals())
if __name__ == "__main__":
    cli()
```

While adding DAG as a comment fixes the problem. This DAG file gets scanned by webserver and scheduler, everything works completely fine:
```
import airflow
# DAG
import company.airflow.etlengine2 as engine

cli = engine.bootstrap(globals())
if __name__ == "__main__":
    cli()
```

One thing to note is that adding the comment still requires a restart of scheduler for the DAGs to appear in the UI.

Also, setting `dag_discovery_safe_mode = False` mitigates the issue.

### What you think should happen instead

According to documentation and the config file, both `airflow` and `DAG` should be keywords for safe DAG file loading.

```
# When discovering DAGs, ignore any files that don't contain the strings ``DAG`` and ``airflow``.
dag_discovery_safe_mode = True
```

### How to reproduce

Create a DAG file without using the keyword `DAG` in it while having `dag_discovery_safe_mode = True` in your Airflow settings file.

### Operating System

macOS 12.6.3

### Versions of Apache Airflow Providers

apache-airflow-providers-amazon==7.3.0
apache-airflow-providers-common-sql==1.3.4
apache-airflow-providers-ftp==3.3.1
apache-airflow-providers-hashicorp==3.3.0
apache-airflow-providers-http==4.2.0
apache-airflow-providers-imap==3.1.1
apache-airflow-providers-mysql==4.0.2
apache-airflow-providers-slack==7.2.0
apache-airflow-providers-sqlite==3.3.1

### Deployment

Other

### Are you willing to submit PR?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by boring-cyborg** [comment]
Thanks for opening your first issue here! Be sure to follow the issue template! If you are willing to raise PR to address this issue please do so, no need to wait for approval.

**Comment by potiuk** [comment]
I do not understand what the issue is. for me the descrption of the config is precise and correct. If you think it can be improved - feel free to open PR with a proposal rather than opening an issue.

**Comment by potiuk** [comment]
(converted to a discussion if needed).

**Comment by andrius-paurys** [comment]
Sorry, only after looking at ``airflow.utils.file.might_contain_dag()`` I've realized that both of the keywords are required. Well, I'll admit that I wasn't paying attention, because docs formulate the condition without confusion, but the config file comment I think is a bit misleading.

> ignore any files that don't contain the strings ``DAG`` and ``airflow``

I would understand this sentence as it ignoring files that don't contain both, `DAG` and `airflow`. Meaning if it contains at least one of them, the condition should not be met and the file should not be skipped.

**Reply by o-nikolas** [reply]
Yeah, I can see that confusion in the wording. Care to create a PR with an update to the comment to clarify that both need to be present not just one or the other?

**Reply by potiuk** [reply]
Yes. Would be great to formulate it in a better way  - so adding a PR to improve that is a good idea @andrius-paurys .

**Reply by andrius-paurys** [reply]
@o-nikolas @potiuk sorry for the delay. Here's the PR clarifying the comment: https://github.com/apache/airflow/pull/30459
